js/background.js

```diff
@@ -483,84 +499,93 @@
-                id = data.source;
-            }
-
-            return ConversationController.getOrCreateAndWait(id, type).then(function() {
-                return message.handleDataMessage(data.message, ev.confirm, {
-                    initialLoadComplete: initialLoadComplete
-                });
-            });
-        });
-    }
-
-    function onSentMessage(ev) {
-        var now = new Date().getTime();
-        var data = ev.data;
-
-        var type, id;
-        if (data.message.group) {
-            type = 'group';
-            id = data.message.group.id;
-        } else {
-            type = 'private';
-            id = data.destination;
-        }
-
-        if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
-            return ConversationController.getOrCreateAndWait(id, type).then(function(convo) {
-              return convo.save({profileSharing: true}).then(ev.confirm);
-            });
-        }
-
-        var message = new Whisper.Message({
-            source         : textsecure.storage.user.getNumber(),
-            sourceDevice   : data.device,
-            sent_at        : data.timestamp,
-            received_at    : now,
-            conversationId : data.destination,
-            type           : 'outgoing',
-            sent           : true,
-            expirationStartTimestamp: data.expirationStartTimestamp,
-        });
+  /* eslint-enable */
+  /* jshint ignore:start */
+  function createMessageHandler(
+    { createMessage, getMessageDescriptor, handleProfileUpdate }
+  ) {
+    return async (event) => {
+      const { data, confirm } = event;
+
+      const messageDescriptor = getMessageDescriptor(data);
+
+      const isProfileUpdate = Boolean(
+        // eslint-disable-next-line no-bitwise
+        data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE
+      );
+      if (isProfileUpdate) {
+        return handleProfileUpdate({ data, confirm, messageDescriptor });
+      }
 
-        return isMessageDuplicate(message).then(function(isDuplicate) {
-            if (isDuplicate) {
-                console.log('Received duplicate message', message.idForLogging());
-                ev.confirm();
-                return;
-            }
+      const message = createMessage(data);
+      const isDuplicate = await isMessageDuplicate(message);
+      if (isDuplicate) {
+        console.log('Received duplicate message', message.idForLogging());
+        return event.confirm();
+      }
 
-            return ConversationController.getOrCreateAndWait(id, type).then(function() {
-                return message.handleDataMessage(data.message, ev.confirm, {
-                    initialLoadComplete: initialLoadComplete
-                });
-            });
-        });
-    }
+      const upgradedMessage = await Message.upgradeSchema(data.message);
+      await ConversationController.getOrCreateAndWait(
+        messageDescriptor.id, messageDescriptor.type
```

Comment:
When we've broken parameters out into different lines from the function call, I think each parameter should be on its own line. The comma is hiding in that line.

Reply:
Done: 799c8817633f6afa0b731fc3b5434e463bd850e3